crates/burn-core/src/lr_scheduler/linear.rs

```diff
@@ -0,0 +1,133 @@
+use super::LrScheduler;
+use crate as burn;
+use crate::{config::Config, LearningRate};
+use burn_tensor::backend::Backend;
+
+/// The configuration for creating a linear learning rate scheduler.
+///
+/// This scheduler starts at a learning rate `initial_lr`, then changes the learning rate by a constant amount on each
+/// iteration until reaching a final learning rate `final_lr`. The `num_iters` parameter controls how many iterations
+/// are needed to go from `initial_lr` to `final_lr`.
+#[derive(Config)]
+pub struct LinearLrSchedulerConfig {
+    // The initial learning rate.
+    initial_lr: LearningRate,
+    // The final learning rate.
+    final_lr: LearningRate,
+    // The number of iterations before reaching the final learning rate.
+    num_iters: usize,
+}
+
+impl LinearLrSchedulerConfig {
+    /// Initializes a [linear learning rate scheduler](LinearLrScheduler).
+    ///
+    /// # Panics
+    /// This function panics if `initial_lr` and `final_lr` are not between 0 and 1.
+    pub fn init(&self) -> LinearLrScheduler {
+        assert!(
+            self.initial_lr > 0. && self.initial_lr <= 1.,
+            "Initial learning rate must be greater than 0 and at most 1"
+        );
+        assert!(
+            self.final_lr >= 0. && self.final_lr <= 1.,
+            "Final learning rate must be at least 0 and at most 1"
+        );
+
+        LinearLrScheduler {
+            previous_lr: self.initial_lr,
+            step_size: (self.final_lr - self.initial_lr) / self.num_iters as LearningRate,
```

Comment:
We should also use `as f64` here instead.
`step_size: (self.final_lr - self.initial_lr) / self.num_iters as f64`